Numpy Array - creating numpy arrays

In [3]:
import numpy as np


In [ ]:
# 1D Array
arr = np.array([1, 2, 3, 4, 5])
print("1D NumPy Array:", arr)

1D NumPy Array: [1 2 3 4 5]


In [ ]:
# 2D Array
arr_2d = np.array([[1, 2], [3, 4]])
print("2D NumPy Array:\n", arr_2d)

2D NumPy Array:
 [[1 2]
 [3 4]]


In [ ]:
# 0D, 1D, 2D, 3D Arrays
a = np.array(42)
b = np.array([1, 2, 3, 4, 5])
c = np.array([[1, 2, 3], [4, 5, 6],[7,8,9]])
d = np.array([[[1, 2, 3], [4, 5, 6]], [[1, 2, 3], [4, 5, 6]]])
print("\n ",a)
print("\n",b)
print("\n",c)
print("\n",d)



  42

 [1 2 3 4 5]

 [[1 2 3]
 [4 5 6]
 [7 8 9]]

 [[[1 2 3]
  [4 5 6]]

 [[1 2 3]
  [4 5 6]]]


ndim - Tells you how many dimensions your array has.


In [ ]:
print(a.ndim)  # 0D
print(b.ndim)  # 1D
print(c.ndim)  # 2D
print(d.ndim)  # 3D


0
1
2
3


shape-Tells you how many elements exist in each dimension.
It's like the size of the array in every direction.


In [3]:
arr=np.array([1,2,3])
print(arr.shape)


(3,)


The shape of an array tells us how many dimensions it has and how many elements are in each

In this case, arr is a 1D array with 4 elements, so the shape is (4,). The comma reminds us that it's a tuple — not just a number."

Slicing and Indexing:
It is for extracting a particular set of elements from the array.

In [7]:
# Simple Indexing
arr = np.array([[1,2,3,4,5],  [6,7,8,9,10]])
print('2nd element on 1st row:', arr[1, 4]) #arr[row,column]



2nd element on 1st row: 10


In [13]:
# 3D Indexing
arr = np.array([
    [[1, 2, 3], [4, 5, 6]], 
    [[7, 8, 9], [10, 11, 12]]
])
print(arr[0, 1, 0]) # arr[axis,row,column]


4


In [17]:
# 1D Slicing
arr = np.array([1, 2, 3, 4, 5, 6, 7])
print(arr[3:5])


[4 5]


Array Initialization Methods- creating an array by giving starting no, end no and the  increment


In [19]:
# Using arange (Start, Stop, Step)
arr = np.arange(40, 55, 3)
print("Array of numbers:", arr)
#two dimensional array
print(arr.shape)
arr_1 =np.arange(1,17).reshape(8,2)
print(arr_1)


Array of numbers: [40 43 46 49 52]
(5,)
[[ 1  2]
 [ 3  4]
 [ 5  6]
 [ 7  8]
 [ 9 10]
 [11 12]
 [13 14]
 [15 16]]


In [27]:
#create 3d array using arange function
arr = np.arange(0, 12, 2)
print(arr)
print("Shape before reshape:", arr.shape)

arr3d = arr.reshape(1, 2, 3)
print("Shape after reshape:", arr3d.shape)
print(arr3d)
print(arr3d[0, 1, 2])

[ 0  2  4  6  8 10]
Shape before reshape: (6,)
Shape after reshape: (1, 2, 3)
[[[ 0  2  4]
  [ 6  8 10]]]
10


Using linspace (Evenly spaced) -- linspace means "linear space".


It generates evenly spaced numbers over a specified interval.




np.linspace(start, stop, num)- ---syntax(num-no of values)


NumPy calculates the step size automatically.
Step size=(stop−start​) / num-1


In [35]:
# Using linspace (Evenly spacd)
arr = np.linspace(0, 2, 8)
print(arr)


[0.         0.28571429 0.57142857 0.85714286 1.14285714 1.42857143
 1.71428571 2.        ]


In [41]:
# Random values
arr = np.random.rand(2, 2)
print(arr)


[[0.55561297 0.60669108]
 [0.89575187 0.45793476]]


rand() → Only floats in [0, 1)-generates a 2×2 array (matrix) filled with random numbers between 0 and 1.

np.random.randn()	- Normal (Gaussian) distribution	Covers negative and positive values(not evenly distributed)

In [51]:
import numpy as np

arr = np.random.randn(2, 3)
print(arr)


[[-0.20210138  1.62937932  0.72156234]
 [ 0.04445747  1.29656757 -1.29167653]]


uniform-Use uniform(low, high, size=...) for floats

In [47]:
arr = np.random.uniform(20, 30, size=(2, 2))
print(arr)
print(arr.dtype)

[[29.5883875  21.79715091]
 [25.53470309 23.35335086]]
float64


randint- Use randint(low, high, size=...) for integers

In [13]:
arr = np.random.randint(10, 15, size=(3, 3))
print(arr)
print(arr.size)


[[10 12 11]
 [11 13 14]
 [13 12 10]]
9


In [ ]:
# Array Inspection
arr = np.array([[1, 2], [3, 4]])
print(arr.shape)  # Shape (rows, columns)



(2, 2)


In [ ]:
arr = np.array([1, 2, 3])
print(arr.dtype)  # Data type


int64


In [ ]:
arr = np.array([[1, 2, 3], [4, 5, 6]])
print(arr.size)  # Total number of elements


6


In [ ]:
#Array Operations & Broadcasting
a = np.array([10, 20, 30])
b = np.array([1, 2, 3])

print("Addition:", a + b)  # Element-wise addition
print("Multiplication by scalar:", a * 2)
print("Element-wise multiplication:", a * b)
print("Mean of array a:", np.mean(a))



Addition: [11 22 33]
Multiplication by scalar: [20 40 60]
Element-wise multiplication: [10 40 90]
Mean of array a: 20.0


In [ ]:
# Universal Functions (ufuncs)

arr = np.array([1, 4, 9])
print(np.sqrt(arr))  # Square root

[1. 2. 3.]


In [ ]:
arr = np.array([0, np.pi/2])
print(np.sin(arr))  # Sine


[0. 1.]


In [ ]:
arr = np.array([1, 2])
print(np.exp(arr))  # Exponential

[2.71828183 7.3890561 ]


In [ ]:
import numpy as np

np.floor()
🔹 Rounds down to the nearest integer.

In [ ]:


arr = np.array([1.2, 2.8, -3.7])
print(np.floor(arr))


[ 1.  2. -4.]


np.ceil()
🔹 Rounds up to the nearest integer.

In [ ]:
arr = np.array([1.2, 2.8, -3.7])
print(np.ceil(arr))
print("trunc",np.trunc(arr))#removes only the decimal
print("round",np.round(arr))

[ 2.  3. -3.]
trunc [ 1.  2. -3.]
round [ 1.  3. -4.]


np.log()
🔹 Returns the natural logarithm (base e) of each element.-[1, e, e²]-

np.e → This is Euler's number ≈ 2.718

np.e**2 → e² ≈ 7.389

arr = [1.0, 2.718..., 7.389...]

In [ ]:
arr = np.array([1, np.e, np.e**2])
print(np.log(arr))


[0. 1. 2.]
